src/openforms/utils/tests/html_assert.py

```diff
@@ -0,0 +1,57 @@
+import re
+
+from django.test import SimpleTestCase
+
+
+class HTMLAssertMixin:
+    def assertTagWithTextIn(self, tag, text, document_str):
+        """
+        check for html tags and their content while ignoring tag attributes
+        """
+        if not re.search(f"<{tag}(?: [^>]*)?>\s*{text}", document_str):
+            self.fail(f"cannot find <{tag}..>{text} in: {document_str}")
+
+    def assertNotTagWithTextIn(self, tag, text, document_str):
+        """
+        check for html tags and their content while ignoring tag attributes
+        """
+        if re.search(f"<{tag}(?: [^>]*)?>\s*{text}", document_str):
+            self.fail(f"unexpectedly found <{tag}..>{text} in: {document_str}")
+
+
+class HTMLAssertMixinTest(HTMLAssertMixin, SimpleTestCase):
```

Comment:
I'm not sure if these tests are picked up because the python module does not start with `test_`:

```
System check identified no issues (0 silenced).
test_allow_redirect_url (openforms.utils.tests.test_redirect.RedirectTests) ... ok
test_base_url_with_path (openforms.utils.tests.test_urls.BuildAbsoluteUriTests) ... ok
test_basic (openforms.utils.tests.test_urls.BuildAbsoluteUriTests) ... ok
test_basic_with_port (openforms.utils.tests.test_urls.BuildAbsoluteUriTests) ... ok
test_basic_with_request (openforms.utils.tests.test_urls.BuildAbsoluteUriTests) ... ok
test_invalid_bsns (openforms.utils.tests.test_validators.BSNValidatorTestCase) ... ok
test_valid_bsns (openforms.utils.tests.test_validators.BSNValidatorTestCase) ... ok
test_invalid_bsns (openforms.utils.tests.test_validators.RSINValidatorTestCase) ... ok
test_valid_bsns (openforms.utils.tests.test_validators.RSINValidatorTestCase) ... ok
test_json_logic_processing_simple (openforms.utils.tests.test_json_logic.JSONLogicUtilsTests) ... ok
test_nested_json_logic (openforms.utils.tests.test_json_logic.JSONLogicUtilsTests) ... ok
test_simple_json_schema_with_required (openforms.utils.tests.test_jsonschema_integration.CamelizeTests) ... ok
test_simple_json_schema_without_required (openforms.utils.tests.test_jsonschema_integration.CamelizeTests) ... ok
test_simple_unmodified (openforms.utils.tests.test_jsonschema_integration.CamelizeTests) ... ok

----------------------------------------------------------------------
Ran 14 tests in 0.036s

OK
```